server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1902,7 +1902,7 @@ protected void mtuCheckForVpcNetwork(Long vpcId, Pair<Integer, Integer> interfac
             }
             s_logger.warn(String.format("VPC public MTU already set at VPC creation phase to: %s. Ignoring public MTU " +
                     "passed during VPC network tier creation ", vpc.getPublicMtu()));
-            interfaceMTUs.set(vpc.getPublicMtu(), privateMtu);
+            interfaceMTUs.set(vpc.getPublicMtu(), interfaceMTUs.second());
```

Comment:
I think we can do with some more cleanup in this method. this method is passed a pair of MTUs and a public and a private MTU.

Reply:
@DaanHoogland 
`publicMtu` is the public MTU passed by API
the pair `interfaceMTUs` is the final public/private MTUs to be used

`privateMtu` is the MTU private passed by API, it is not used in this method, I will remove it